tests/basic.rs

```diff
@@ -185,3 +185,18 @@ fn drop_value() {
     // Second NotifyOnDrop was dropped when the map was dropped
     assert_eq!(Arc::strong_count(&dropped2), 1);
 }
+
+#[test]
+// Test that same values exists in both maps (original and cloned)
+fn clone_map() {
```

Comment:
It might be good to take a look through the various Java tests and see if any of those test the constructor that copies an existing map.

Reply:
This is what they are testing. I added the part in where they test they are not mapping the same tables.

``` java
    /**
     * ConcurrentHashMap(map) creates a new map with the same mappings
     * as the given map
     */
    public void testConstructor5() {
        ConcurrentHashMap map1 = map5();
        ConcurrentHashMap map2 = new ConcurrentHashMap(map5());
        assertTrue(map2.equals(map1));
        map2.put(one, "F");
        assertFalse(map2.equals(map1));
    }
```

Reply:
Yeah, changing one and then making sure that the cloned-from map does not also change seems like a good thing to test.